test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/CompleteTagHelperDescriptorComparer.cs

```diff
@@ -23,16 +23,22 @@ bool IEqualityComparer<TagHelperDescriptor>.Equals(TagHelperDescriptor descripto
             return base.Equals(descriptorX, descriptorY) &&
                    // Tests should be exact casing
                    string.Equals(descriptorX.TagName, descriptorY.TagName, StringComparison.Ordinal) &&
-                   descriptorX.Attributes.SequenceEqual(descriptorY.Attributes,
-                                                        CompleteTagHelperAttributeDescriptorComparer.Default);
+                   Enumerable.SequenceEqual(
+                       descriptorX.RequiredAttributes,
+                       descriptorY.RequiredAttributes,
+                       StringComparer.OrdinalIgnoreCase) &&
+                   descriptorX.Attributes.SequenceEqual(
+                       descriptorY.Attributes,
+                        CompleteTagHelperAttributeDescriptorComparer.Default);
         }
 
         int IEqualityComparer<TagHelperDescriptor>.GetHashCode(TagHelperDescriptor descriptor)
         {
-            return HashCodeCombiner.Start()
-                                   .Add(base.GetHashCode())
-                                   .Add(descriptor.Attributes)
-                                   .CombinedHash;
+            return HashCodeCombiner
+                .Start()
+                .Add(base.GetHashCode())
+                .Add(descriptor.Attributes)
+                .CombinedHash;
         }
 
         private class CompleteTagHelperAttributeDescriptorComparer : IEqualityComparer<TagHelperAttributeDescriptor>
```

Comment:
nit since this should go away: name is not correct because you're now ignoring a number of properties the base (incomplete?) `TagHelperDescriptorComparer`'s inner class would check